packages/protocol/contracts/governance/Election.sol

```diff
@@ -678,40 +716,64 @@ contract Election is
    * @param account The address of the voting account.
    * @param value The number of votes.
    */
-  function decrementActiveVotes(address group, address account, uint256 value) private {
+  function decrementActiveVotes(address group, address account, uint256 value)
+    private
+    returns (uint256)
+  {
     ActiveVotes storage active = votes.active;
     active.total = active.total.sub(value);
 
-    // Rounding may cause getActiveVotesUnitsDelta to return 0 for value != 0, preventing users
-    // from revoking the last of their votes. The case where value == activeVotes is special cased
+    // Rounding may cause votesToUnits to return 0 for value != 0, preventing users
+    // from revoking the last of their votes. The case where value == votes is special cased
     // to prevent this.
-    uint256 unitsDelta = 0;
+    uint256 units = 0;
     uint256 activeVotes = getActiveVotesForGroupByAccount(group, account);
     GroupActiveVotes storage groupActive = active.forGroup[group];
     if (activeVotes == value) {
-      unitsDelta = groupActive.unitsByAccount[account];
+      units = groupActive.unitsByAccount[account];
     } else {
-      unitsDelta = getActiveVotesUnitsDelta(group, value);
+      units = votesToUnits(group, value);
     }
 
     groupActive.total = groupActive.total.sub(value);
-    groupActive.totalUnits = groupActive.totalUnits.sub(unitsDelta);
-    groupActive.unitsByAccount[account] = groupActive.unitsByAccount[account].sub(unitsDelta);
+    groupActive.totalUnits = groupActive.totalUnits.sub(units);
+    groupActive.unitsByAccount[account] = groupActive.unitsByAccount[account].sub(units);
+    return units;
   }
 
   /**
-   * @notice Returns the delta in active vote denominator for `group`.
+   * @notice Returns the number of units corresponding to `value` active votes.
    * @param group The address of the validator group.
-   * @param value The number of active votes being added.
-   * @return The delta in active vote denominator for `group`.
-   * @dev Preserves unitsDelta / totalUnits = value / total
+   * @param value The number of active votes.
+   * @return The corresponding number of units.
    */
-  function getActiveVotesUnitsDelta(address group, uint256 value) private view returns (uint256) {
+  function votesToUnits(address group, uint256 value) private view returns (uint256) {
+    FixidityLib.Fraction memory fixedValue = FixidityLib.newFixed(value);
     if (votes.active.forGroup[group].total == 0) {
-      return value;
+      return fixedValue.unwrap();
+    } else {
+      return
+        fixedValue.multiply(FixidityLib.wrap(votes.active.forGroup[group].totalUnits)).unwrap().div(
+          votes.active.forGroup[group].total
```

Comment:
What is the difference between this and `value.mul(votes.active.forGroup[group].totalUnits).div(votes.active.forGroup[group].total)` ?
Anyway it looks like we are getting close to overflow, for example if we have 1e9 cGLD (1e27 wei), this gives 1e51 units, then when they are multiplied, we have 1e78 (overflow).

Reply:
Good catch, reduced to precision to 1e20